lib/publiccloud/azure.pm

```diff
@@ -72,18 +73,45 @@ sub find_img {
     my $container = $self->container;
     my $storage_account = get_var('PUBLIC_CLOUD_STORAGE_ACCOUNT', 'eisleqaopenqa');
     my $key = $self->get_storage_account_keys($storage_account);
-    my $cmd_show = "az storage blob show --account-key $key -o json " .
-      "--container-name '$container' --account-name '$storage_account' --name '$name' " .
-      '--query="{name: name,createTime: properties.creationTime,md5: properties.contentSettings.contentMd5}"';
-    record_info('BLOB INFO', script_output($cmd_show, proceed_on_failure => 1));
-
-    my $json = script_output("az image show --resource-group " . $self->resource_group . " --name $name", 60, proceed_on_failure => 1);
-    record_info('IMG INFO', $json);
-    my $image;
+
+    $json = script_output("az storage blob show --account-key $key -o json " .
+          "--container-name '$container' --account-name '$storage_account' --name '$name' " .
+          '--query="{name: name,createTime: properties.creationTime,md5: properties.contentSettings.contentMd5}"',
+        proceed_on_failure => 1);
+    record_info('BLOB INFO', $json);
     eval {
-        $image = decode_azure_json($json)->{name};
+        $md5 = decode_azure_json($json)->{md5};
     };
-    record_info('IMG NOT-FOUND', "Cannot find image $name. Need to upload it.\n$@") if ($@);
+    if ($@) {
+        record_info('BLOB NOT-FOUND', "Cannot find blob $name. Need to upload it.\n$@");
+    } elsif (!$md5 || $md5 !~ /^[a-fA-F0-9]{32}$/) {
+        record_info('INVALID', "The blob $name does not have valid md5 field.");
+        return $image;
+    }
+
+    my $arch = (check_var('PUBLIC_CLOUD_ARCH', 'arm64')) ? 'Arm64' : 'x64';
```

Comment:
ha funny , never noticed that before . I have seen "x86" and "x86_64" and "x86_32" but I never seen "x64" Microsoft is really doing everything in its own way ...

Reply:
ofc. this comment does not require any reaction :)